File: 25.py
converter = {
    "1":[",",".","?","!"],
    "2":["a","b","c"],
    "3":["d","e","f"],
    "4":["g","h","i"],
    "5":["j","k","l"],
    "6":["m","n","o"],
    "7":["p","q","r","s"],
    "8":["t","u","v"],
    "9":["w","x","y","z"]
}
import re
def t9_conversor(code):
    if type(code)==str and re.findall(r"[1-9]{1,4}(?!>-)",code)==code.split("-"): #si es una cadena y al partirlo usando - quedan las coincidencias con números de a lo más 4 digitos
        code = code.split("-") #lo separamos por el -
        conversion = "" #creamos una variable para la conversión
        for element in code:
            if element[0] not in ["1","7","9"] and len(element)==4: #si se pulsan 4 veces algo distinto a 1,7 o 9
                print("los únicos números que admiten 4 pulsaciones son 1, 7 y 9") #informamos del error
                return None
            if element.count(element[0])!=len(element): #si no esta conformado por un solo número
                print("solo debe pulsarse un número una o más veces seguidas antes de un guio: 333-1 correcto |  332-1 incorrecto")
                return False
            conversion+=converter[element[0]][element.count(element[0])-1] #si supera las condiciones, añadimos el caracter según la cantidad de pulsaciones
        else:
            return conversion #en caso de acabar el for, regresa la conversión
    else:
        print("tu entrada deben ser grupos con, una a cuatro pulsaciones de número y un guion, salvo el último, ejemplo: 22-3")
        return False
    
print(t9_conversor("6-666-88-777-33-3-33-888")) #probamos el ejemplo
print(t9_conversor("22-777-444-2-66-7-2-777-777-2"))

File: 25.js
"use strict";
const t9Conversor = t9string =>{
    try{
        t9string = t9string.split("-");
        let conversion={
            "1":[",",".","?","!"],
            "2":["A","B","C"],
            "3":["D","E","F"],
            "4":["G","H","I"],
            "5":["J","K","L"],
            "6":["M","N","O"],
            "7":["P","Q","R","S"],
            "8":["T","U","V"],
            "9":["W","X","Y","Z"],
            "0":[" "]
        }
        let conversed = "";
        for(let item of t9string){
            if(conversion[item[0]]==undefined){
                console.log("debes usar solo números del 0 al 9");
                return false;
            }
            if(item!="".padEnd(item.length,item[0])){
                console.log("Debes poner el mismo número seguido");
                return false;
            }else{
                let i = item.length-1%conversion[item[0]].length; //se reinicia si se pulsa de más.
                conversed+=conversion[item[0]][i];
                console.log(i);
            }
        }
        return conversed;
    }catch{
        console.log("asegurate de introducir una cadenas, además debes poner números del 0 al 9 separados por -, ej: 22-44-999");
        return null;
    }
}